internal/server/proto/server.proto

```diff
@@ -1854,6 +1854,9 @@ message Deployment {
   // id is the unique ID for this deployment
   string id = 1;
 
+  // url is the URL to the Deployment
+  string url = 18;
```

Comment:
This is fine, but as a bigger comment I think we should rethink our data model slightly around URLs. Right now it is unclear what the difference is between `deploy.Url` and `deploy.Preload.Url` and both MIGHT be empty or both MIGHT NOT be empty or ONE may be empty, etc. Actionably, I think we should:

* Have more detailed comments.
* Have a canonical URL field that has SOME URL if one exists (may still be empty if none exist)
* Clearly delineate between URL service URL and Platform URL

Reply:
I didn't found any documentation around `deploy.Preload.Url`. I think it's supposed to be an Horizon pre-defined URL before the application is even deployed, is that correct?

Reply:
Yes, because our data model only understands URL service URLs currently. :) That's the big conceptual shift in this PR I think is that we're now saying deployments may have a platform-provided URL. But the data model doesn't quite reflect that and I think it should.

Reply:
How would you like to approach this?

Do you have any proposed change?